criu/sockets.c

```diff
@@ -465,18 +466,33 @@ int do_restore_opt(int sk, int level, int name, void *val, int len)
 	return 0;
 }
 
-static int sk_setbufs(void *arg, int fd, pid_t pid)
+int sk_setbufs(int sk, uint32_t *bufs)
 {
-	u32 *buf = (u32 *)arg;
+	uint32_t sndbuf = bufs[0], rcvbuf = bufs[1];
 
-	if (restore_opt(fd, SOL_SOCKET, SO_SNDBUFFORCE, &buf[0]))
-		return -1;
-	if (restore_opt(fd, SOL_SOCKET, SO_RCVBUFFORCE, &buf[1]))
-		return -1;
+	if (restore_opt(sk, SOL_SOCKET, SO_SNDBUFFORCE, &sndbuf) ||
+	    restore_opt(sk, SOL_SOCKET, SO_RCVBUFFORCE, &rcvbuf)) {
```

Comment:
sk_setbufs->restore_opt->do_restore_opt has an error printing. In case of fallback printing errors may be confusing. Maybe it's better to use setsockopt here directly to have full control over error printing?

Reply:
Agreed, fixed as you suggested.